src/basic/unit-file.c

```diff
@@ -87,8 +88,10 @@ int unit_validate_alias_symlink_or_warn(int log_level, const char *filename, con
          * -ELOOP for an alias to self.
          */
 
-        src = basename(filename);
-        dst = basename(target);
+        path_extract_filename(filename, &temp);
```

Comment:
this leaks the buffers, it needs a variable annotated with _cleanup_free_, also there's no need for temp, just use src and dst directly

Reply:
Thank you sir. Sorry for making you have to review again.

src and DST are const char *,  so where I tried using them directly, the compiler wouldn't run the code
based on the fact that path_extract_filename() expects the address of a char * not a const char *.

Is there a way around this sir?

Reply:
When I tried using a _clean_up_free_ variable, the situation was worse. It was more than 10 test failures before I cancelled. Lots of the failures were build tests on logind.session or so.

Looked like the variable got freed before they could be accessed.

The major issue though is the const char* vs char* sir.

Reply:
remove the const - you'll need to adjust a few things

Reply:
> remove the const - you'll need to adjust a few things

Thank you so much sir.

I'll push an update as soon as I get back.

Reply:
I'm sorry I couldn't push an update on this again today sir. I'll do before noon tomorrow.

I'm assuming our noons are thesame.